### PR TITLE
Updated the documentation of modules to match argspec including the t…

### PR DESCRIPTION
**Pull request by grastogi23**
…ype setting for bool parameters

##### SUMMARY

update documentation of Avi modules to conform to new guidelines.

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

network/avi

##### ANSIBLE VERSION

```
ansible 2.6.0 (grastogi-br1 a982fafb85) last updated 2018/02/27 22:36:20 (GMT -700)
  config file = /home/grastogi/.ansible.cfg
  configured module search path = [u'/home/grastogi/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/grastogi/workspace/ansible/lib/ansible
  executable location = /home/grastogi/workspace/ansible/bin/ansible
  python version = 2.7.6 (default, Jun 22 2015, 17:58:13) [GCC 4.8.2]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by grastogi23** [comment]
@gundalow Pl. review and approve the PR.

File: lib/ansible/modules/network/avi/avi_applicationprofile.py [hunk 1]
```diff
@@ -63,6 +63,14 @@
             - Specifies if client ip needs to be preserved for backend connection.
             - Not compatible with connection multiplexing.
             - Default value when not specified in API or module is interpreted by Avi Controller as False.
+        type: bool
+    preserve_client_port:
+        description:
+            - Specifies if we need to preserve client port while preseving client ip for backend connections.
+            - Field introduced in 17.2.7.
+            - Default value when not specified in API or module is interpreted by Avi Controller as False.
+        version_added: "2.6"
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_applicationprofile.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_applicationprofile.py [hunk 2]
```diff
@@ -175,6 +183,7 @@ def main():
         http_profile=dict(type='dict',),
         name=dict(type='str', required=True),
         preserve_client_ip=dict(type='bool',),
+        preserve_client_port=dict(type='bool',),
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_applicationprofile.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 3]
```diff
@@ -647,6 +682,7 @@ def main():
         disable_tso=dict(type='bool',),
         disk_per_se=dict(type='int',),
         distribute_load_active_standby=dict(type='bool',),
+        enable_hsm_priming=dict(type='bool',),
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_serviceenginegroup.py [hunk 4]
```diff
@@ -179,19 +192,32 @@
         description:
             - Use both the active and standby service engines for virtual service placement in the legacy active standby ha mode.
             - Default value when not specified in API or module is interpreted by Avi Controller as False.
+        type: bool
+    enable_hsm_priming:
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_serviceenginegroup.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_virtualservice.py [hunk 5]
```diff
@@ -473,6 +505,7 @@ def main():
         availability_zone=dict(type='str',),
         avi_allocated_fip=dict(type='bool',),
         avi_allocated_vip=dict(type='bool',),
+        bulk_sync_kvcache=dict(type='bool',),
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_virtualservice.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_virtualservice.py [hunk 6]
```diff
@@ -504,6 +537,7 @@ def main():
         ign_pool_net_reach=dict(type='bool',),
         ip_address=dict(type='dict',),
         ipam_network_subnet=dict(type='dict',),
+        l4_policies=dict(type='list',),
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_virtualservice.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_virtualservice.py [hunk 7]
```diff
@@ -82,11 +84,23 @@
...
             - Field deprecated in 17.1.1.
             - Default value when not specified in API or module is interpreted by Avi Controller as False.
+        type: bool
     avi_allocated_vip:
         description:
             - (internal-use) vip allocated by avi in the cloud infrastructure.
             - Field deprecated in 17.1.1.
             - Default value when not specified in API or module is interpreted by Avi Controller as False.
+        type: bool
+    bulk_sync_kvcache:
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_virtualservice.py]: Please don't add new options in this PR, please remove this

File: lib/ansible/modules/network/avi/avi_virtualservice.py [hunk 8]
```diff
@@ -217,10 +239,16 @@
         description:
             - Subnet and/or network for allocating virtualservice ip by ipam provider module.
             - Field deprecated in 17.1.1.
+    l4_policies:
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_virtualservice.py]: Please don't add new options in this PR, please remove this

**Review by gundalow — CHANGES_REQUESTED** [review]
There are a few new options in here that need removing before I can cherry-pick this into `stable-2.5`

Once this has been merged you can raise a new PR to add the new 2.6 features

**Comment by grastogi23** [comment]
@gundalow updated the PR such that new options have been removed.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thanks. Merged into `devel` and cherry-picked into `stable-2.5`

https://github.com/ansible/ansible/pull/36860

You can now raise a fresh PR to add in the new Ansible 2.6 features

**Pull request merged**